pkg/webhook/interface.go

```diff
@@ -0,0 +1,15 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package webhook
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Conversion interface {
```

Comment:
Along the lines of my prior comment wrt `controller-runtime` - this interface is nearly identical to its `Convertible` interface.

Reply:
correct. To summarise the difference:

- The conversion functions you see across the APIs(alpha,beta), will still fit if we satisfy the controller-runtime interface.
- The additional logic we require (as a consequence of not using the controller-runtime interface), is to handle in our webhook the `ConversionReview` request, and  return a response  with a `ConversionReview` containing a `convertedObjects` list.